include/highfive/H5DataType.hpp

```diff
@@ -83,6 +83,8 @@ class DataType : public Object {
     /// \brief Returns whether the type is a Reference
     bool isReference() const;
 
+    DataType() = default;
+    DataType(hid_t hid) : Object(hid) {};
```

Comment:
I find this unsafe and not required for your use case.

Reply:
done